File: pset6/caesar.py
# Implements caesar cipher
# Usage: python caesar.py n

import sys
from cs50 import get_string

def main():
    if len(sys.argv) != 2:
        sys.exit("Usage: python caesar.py n")

    key = int(sys.argv[1])
    if key == 0:
        sys.exit("Usage: python caesar.py n")

    plain = get_string("plaintext: ")
    plainL = len(plain)
    print("ciphertext: ")

    for c in plain:
        ch = ord(c)
        if c.isupper():
            print(chr(((ch - 65 + key) % 26) + 65), end="")
        elif c.islower():
            print(chr(((ch - 97 + key) % 26) + 97), end="")
        else:
            print(chr(ch), end="")
    print()


if __name__ == "__main__":
    main()

File: pset2/caesar.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int main(int argc, string argv[])
{
    if (argc == 2)
    {
        int key = atoi(argv[1]);
        if (key != 0)
        {
            string plain = get_string("plaintext: ");
            int plainL = strlen(plain);
            printf("ciphertext: ");
            for (int i = 0; i < plainL; i++)
            {
                char c = plain[i];
                if (isupper(c))
                {
                    printf("%c", ((c - 65 + key) % 26) + 65);
                }
                else if (islower(c))
                {
                    printf("%c", ((c - 97 + key) % 26) + 97);
                }
                else
                {
                    printf("%c", c);
                }
            }
            printf("\n");
            return 0;
        }
        else
        {
            printf("Usage: ./caesar key\n");
            return 1;
        }
    }
    else
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }
}
